crates/bevy_reflect/bevy_reflect_derive/src/impls/structs.rs

```diff
@@ -106,38 +110,38 @@ pub(crate) fn impl_struct(reflect_struct: &ReflectStruct) -> TokenStream {
         #typed_impl
 
         impl #impl_generics #bevy_reflect_path::Struct for #struct_name #ty_generics #where_clause {
-            fn field(&self, name: &str) -> Option<&dyn #bevy_reflect_path::Reflect> {
+            fn field(&self, name: &str) -> #option<&dyn #bevy_reflect_path::Reflect> {
                 match name {
-                    #(#field_names => Some(&self.#field_idents),)*
-                    _ => None,
+                    #(#field_names => #option::Some(&self.#field_idents),)*
```

Comment:
Replacing `#option` with `#FQOption` results in a compilation error stating:
```
error[E0530]: let bindings cannot shadow unit structs
   --> /home/elbert/dev/bevy/crates/bevy_reflect/bevy_reflect_derive/src/impls/structs.rs:116:40
    |
6   | use crate::fq_std::FQOption;
    |     ----------------------- the unit struct `FQOption` is imported here
...
116 |                     #(#field_names => #FQOption::Some(&self.#field_idents),)*
    |                                        ^^^^^^^^ cannot be named the same as a unit struct

For more information about this error, try `rustc --explain E0530`.
```

This error only occurs inside the repetition syntax `#( ... )*` of proc macros inside match statements. It works fine everywhere else. Just for info, `FQOption` is declared in a new file named fq_std.rs.
I am not able to understand the error or solve it.

Reply:
Nvm, I realised that this is because the `quote` crate does not allow structs in its repeat syntax. To do this, we would have to use the newer `proc_quote` crate. So I decided to use a local variable for these cases only.